src/SoftRobots/component/constraint/model/CableModel.cpp

```diff
@@ -41,12 +40,66 @@ namespace component
 namespace constraintset
 {
 
-using namespace sofa::defaulttype;
+
+using sofa::defaulttype::Vec2Types;
+using sofa::defaulttype::Vec3Types;
+using sofa::type::Vector2;
+
+template<>
+void CableModel<Vec2Types>::getPositionFromTopology(Coord& position, const int& index)
+{
+    position.x() = l_surfaceTopology.get()->getPX(index);
```

Comment:
I think that whenever you want to use `getPositionFromTopology`, you could actually use something like:
```cpp
DataTypes::getCPos(this->model->mstate->read(core::ConstVecCoordId::position())->getValue()[index]
```
It makes me think that `getPositionFromTopology` is therefore unneeded: to be removed.

However, my suggestion is not efficient. You should get the position vector first, outside of the loop, and then access the cached vector inside the loop.